### PR TITLE
[NUI] Text: Add more scroll functinality to texteditor

### PR DESCRIPTION
**Pull request by Ali-Alzyoud**
### Description of Change ###
1- add Horizontal_Scroll_position property (float)
2- add VerticalScrollPosition property (float)
3- add ScrollBy function (take Vector2)

### API Changes ###

 - ACR:

### CODE REVIEW

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs [hunk 1]
```diff
@@ -1045,6 +1046,44 @@ public bool EnableEditing
...
+        /// <summary>
+        /// Specify horizontal scroll position in text control.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public int HorizontalScrollPosition
+        {
+            get
+            {
+                int temp;
+                GetProperty(TextEditor.Property.HORIZONTAL_SCROLL_POSITION).Get(out temp);
```
Comment (id213sin) [inline src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs:1058]: I want to suggest to fix this code for avoiding CA2000 build warnings: "Call System.IDisposable.Dispose on object created by ~~"

Just like:
                using (PropertyValue propertyValue = GetProperty(TextEditor.Property.LINE_COUNT))
                {
                    propertyValue.Get(out temp);
                }
                
                return temp;

or
just call Dispose or Close at end of code.

**Review by AdunFang — APPROVED** [review]
It is OK.

**Comment by JoogabYun** [comment]
It is Ok

**Review by Seoyeon2Kim — APPROVED** [review]
LGTM. After updating a branch and getting more reviews if you want, you can merge this PR.
Dear @hyunjushin and @huiyueun , could you check it one more time?

**Pull request merged**